src/sentry/crons/consumers/checkin.py

```diff
@@ -0,0 +1,61 @@
+import logging
+from typing import Mapping
+
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies.abstract import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.processing.strategies.commit import CommitOffsets
+from arroyo.processing.strategies.run_task import RunTask
+from arroyo.types import Commit, Message, Partition
+from django.db import transaction
+
+from sentry.models import CheckInStatus, Monitor, MonitorCheckIn, MonitorStatus
+from sentry.utils import json
+from sentry.utils.dates import to_datetime
+
+logger = logging.getLogger(__name__)
+
+
+def process_message(message: Message[KafkaPayload]) -> None:
+    params = json.loads(message.payload.value)
+    current_datetime = to_datetime(params["date_updated"])
+
+    # TODO: Same as MonitorCheckInDetailsEndpoint.put. Keep in sync or factor out.
```

Comment:
@evanpurkhiser The update code is partially duplicated with the cron details endpoint. I can
factor this out into a special update function on `MonitorCheckIn`. Any preference?

Reply:
Let's keep in sync, we can factor out later.

We're also working on environment support right now #42788 so this code is changing a lot anyway

Reply:
If this code is changing a lot, then we should factor it out rather sooner than later. This is just a function, but at least it'll prevent us from going more out of sync. I'll get that addressed.

Reply:
I agree, I wanted to move monitors into a top-level django app where we can nicely consolidate logic, but I haven't had time to do that quite yet. I think we should have some re-usable stuff soon. The thing I am most interested in is just having everything talking to each other, we can swap out what this piece does easily enough.